Sources/Sentry/SentryBinaryImageCache.m

```diff
@@ -99,6 +99,19 @@ - (NSInteger)indexOfImage:(uint64_t)address
     return -1; // Address not found
 }
 
+- (nullable NSString *)pathForImage:(NSString *)binaryName
+{
+    @synchronized(self) {
+        for (SentryBinaryImageInfo *info in _cache) {
+            if ([[info.name.lastPathComponent lowercaseString]
+                    isEqualToString:binaryName.lowercaseString]) {
```

Comment:
`h`: The inAppInclude is a prefix. We must change this to using `hasPrefix`.
```suggestion
                    hasPrefix:binaryName.lowercaseString]) {
```

The InAppLogic does the same see
https://github.com/getsentry/sentry-cocoa/blob/b0646651e1c1e526071883841a6412e4f75f3690/Sources/Sentry/SentryInAppLogic.m#L35

I think it's a good idea to refactor the code so both this code here and the `InAppLogic` use the same code so we only have to change it once.

Reply:
I didn't notice this behaviour of inAppLogic, but I have a problem with it.

If I add `Awesome` to my inAppLogic it will match the `Awesome` lib and also `Core-Awesome` `Data-Awesome` and `Private-Awesome`.

Reply:
No, it won't. The logic uses `hasPrefix` on the `lastPathComponent`. If you add `Awesome` to your inAppLogic `Core-Awesome`, `Data-Awesome`, and `Private-Awesome` won't be inApp, while `Awesome-Core` would be. I think we should stick to this logic.